File: app/src/main/java/com/xpf/p2p/utils/AppUtil.java
package com.xpf.p2p.utils;

import android.content.Context;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;

import com.xpf.http.logger.XLog;

/**
 * Created by xpf on 2017/12/5 :)
 * GitHub:xinpengfei520
 * Function:
 */
public class AppUtil {

    /**
     * 获取当前应用的版本号
     * 如果找不到对应的应用包信息, 就返回"未知版本":unknown
     *
     * @param context 上下文
     * @return 版本号
     */
    public static String getVersion(Context context) {
        String version = "unknown";
        try {
            if (context != null) {
                Context appContext = context.getApplicationContext();
                PackageManager manager = appContext.getPackageManager();
                PackageInfo packageInfo = manager.getPackageInfo(appContext.getPackageName(), 0);
                version = packageInfo.versionName;
            }
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }
        return version;
    }

    /**
     * 返回当前程序版本号
     */
    public static int getAppVersionCode(Context context) {
        String versionName = "";
        int versioncode = 0;
        try {
            // ---get the package info---
            PackageManager pm = context.getPackageManager();
            PackageInfo pi = pm.getPackageInfo(context.getPackageName(), 0);
            versionName = pi.versionName;
            versioncode = pi.versionCode;
            if (versionName == null || versionName.length() <= 0) {
                return 0;
            }
        } catch (Exception e) {
            e.printStackTrace();
            XLog.e("Exception" + e);
        }
        return versioncode;
    }

    /**
     * 获取本地软件版本号名称
     */
    public static String getAppVersionName(Context ctx) {
        String localVersion = "";
        try {
            PackageInfo packageInfo = ctx.getApplicationContext()
                    .getPackageManager()
                    .getPackageInfo(ctx.getPackageName(), 0);
            localVersion = packageInfo.versionName;
            XLog.d("本软件的版本号。。" + localVersion);
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }
        return localVersion;
    }
}

File: app/src/main/java/com/xpf/p2p/utils/AppUtil.kt
package com.xpf.p2p.utils

import android.content.Context
import android.content.pm.PackageManager
import android.view.WindowManager
import com.xpf.p2p.utils.LogUtils

/**
 * Created by xpf on 2017/12/5 :)
 * GitHub:xinpengfei520
 * Function:
 */
object AppUtil {

    @JvmStatic
    fun getVersion(context: Context?): String {
        var version = "unknown"
        try {
            if (context != null) {
                val appContext = context.applicationContext
                val manager = appContext.packageManager
                val packageInfo = manager.getPackageInfo(appContext.packageName, 0)
                version = packageInfo.versionName
            }
        } catch (e: PackageManager.NameNotFoundException) {
            e.printStackTrace()
        }
        return version
    }

    @JvmStatic
    fun getAppVersionCode(context: Context): Int {
        var versioncode = 0
        try {
            val pm = context.packageManager
            val pi = pm.getPackageInfo(context.packageName, 0)
            val versionName = pi.versionName
            versioncode = pi.versionCode
            if (versionName == null || versionName.isEmpty()) {
                return 0
            }
        } catch (e: Exception) {
            e.printStackTrace()
            LogUtils.e("AppUtil", "Exception$e")
        }
        return versioncode
    }

    @JvmStatic
    fun getAppVersionName(ctx: Context): String {
        var localVersion = ""
        try {
            val packageInfo = ctx.applicationContext
                .packageManager
                .getPackageInfo(ctx.packageName, 0)
            localVersion = packageInfo.versionName
            LogUtils.d("AppUtil", "本软件的版本号。。$localVersion")
        } catch (e: PackageManager.NameNotFoundException) {
            e.printStackTrace()
        }
        return localVersion
    }

    @Suppress("DEPRECATION")
    @JvmStatic
    fun getScreenDisplay(context: Context): IntArray {
        val windowManager = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
        val width = windowManager.defaultDisplay.width
        val height = windowManager.defaultDisplay.height
        return intArrayOf(width, height)
    }
}
